core/src/main/java/com/linecorp/armeria/internal/server/ResponseCompleteException.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.internal.server;
+
+import com.linecorp.armeria.common.CancellationException;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+
+/**
+ * A special {@link CancellationException} that aborts an {@link HttpRequest} after the corresponding
+ * {@link HttpResponse} is completed.
+ */
+public final class ResponseCompleteException extends CancellationException {
```

Comment:
Should we make this class to public API because this class could be exposed to users via `HttpRequest.whenComplete()` and they may want to `catch` the exception and ignore it.

Reply:
Moved it to common and also use it on the client side. 😉